File: 0027-remove-element/0027-remove-element.c


int removeElement(int* nums, int numsSize, int val){
    int i = 0, j;
    for(j = 0; j < numsSize; j++)
    {
        if(nums[j] != val)
        {
            nums[i++] = nums[j];
        }
    }
    return i;
}

File: 0027-remove-element/0027-remove-element.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        if(nums.len() == 0)
        {
            return 0 as i32;
        }
        let mut i : usize = 0;
        for j in 0..nums.len()
        {
            if(nums[j] != val)
            {
                nums[i] = nums[j];
                i = i + 1;
            }
        }
        return i as i32;
    }
}